src/commands/run.js

```diff
@@ -82,19 +85,41 @@ class RunCommand extends ApifyCommand {
         }
 
-        // --max-http-header-size=80000
-        // Increases default size of headers. The original limit was 80kb, but from node 10+ they decided to lower it to 8kb.
-        // However they did not think about all the sites there with large headers,
-        // so we put back the old limit of 80kb, which seems to work just fine.
-        const currentNodeVersion = process.versions.node;
-        const lastSupportedVersion = semver.minVersion(SUPPORTED_NODEJS_VERSION);
-        if (semver.gte(currentNodeVersion, lastSupportedVersion)) {
-            env.NODE_OPTIONS = env.NODE_OPTIONS ? `${env.NODE_OPTIONS} --max-http-header-size=80000` : '--max-http-header-size=80000';
-        } else {
-            warning(`You are running Node.js version ${currentNodeVersion}, which is no longer supported. `
-                + `Please upgrade to Node.js version ${lastSupportedVersion} or later.`);
+        if (packageJsonExists) { // Actor is written in Node.js
+            const serverJsFile = path.join(cwd, 'server.js');
+            const packageJson = await loadJson(packageJsonPath);
+            if ((!packageJson.scripts || !packageJson.scripts.start) && !fs.existsSync(serverJsFile)) {
+                throw new Error('The "npm start" script was not found in package.json. Please set it up for your project. '
+                    + 'For more information about that call "apify help run".');
+            }
+
+            // --max-http-header-size=80000
+            // Increases default size of headers. The original limit was 80kb, but from node 10+ they decided to lower it to 8kb.
+            // However they did not think about all the sites there with large headers,
+            // so we put back the old limit of 80kb, which seems to work just fine.
+            const currentNodeVersion = process.versions.node;
+            const lastSupportedVersion = semver.minVersion(SUPPORTED_NODEJS_VERSION);
+            if (semver.gte(currentNodeVersion, lastSupportedVersion)) {
+                env.NODE_OPTIONS = env.NODE_OPTIONS ? `${env.NODE_OPTIONS} --max-http-header-size=80000` : '--max-http-header-size=80000';
+            } else {
+                warning(`You are running Node.js version ${currentNodeVersion}, which is no longer supported. `
+                    + `Please upgrade to Node.js version ${lastSupportedVersion} or later.`);
+            }
+            await execWithLog(getNpmCmd(), ['start'], { env });
+        } else if (mainPyExists) {
+            const pythonVersion = detectPythonVersion(cwd);
+            if (pythonVersion) {
+                if (isPythonVersionSupported(pythonVersion)) {
+                    const pythonCommand = getPythonCommand(cwd);
+                    await execWithLog(pythonCommand, ['-m', 'src'], { cwd });
+                } else {
+                    error(`Python actors require Python 3.8 or higher, but you have Python ${pythonVersion}!`);
```

Comment:
is there a reason to completely bail on old python versions instead of continuing with the warning, like we do with node?

Reply:
Well, the Python SDK does not support Python 3.7 or lower, it will 100% break. So I'd rather break early here, rather than later during the actual code execution.

With Node we don't break on the lower versions, because people can have old actors written using old versions of Apify SDK, which will function fine on old versions of Node.

If we ever deprecate Python 3.8 support, I think we'll add just a warning here, to continue supporting old actors.